packages/core/src/lib/implementation/compare-scorables.unit.test.ts

```diff
@@ -0,0 +1,320 @@
+        },
+      ],
+      added: [],
+      removed: [],
+    });
+  });
+});
+
+describe('compareAudits', () => {
+  it('should match audits by plugin-scoped slug and check for equal scores and values', () => {
+    expect(
+      compareAudits({
+        before: {
+          plugins: [
+            {
+              slug: 'eslint',
+              title: 'ESLint',
+              audits: [
+                {
+                  slug: 'eqeqeq',
+                  title: 'Require the use of `===` and `!==`',
+                  score: 1,
+                  value: 0,
+                  displayValue: 'passed',
+                },
+                {
+                  slug: 'no-case-declarations',
+                  title: 'Disallow lexical declarations in case clauses',
+                  score: 0,
+                  value: 12,
+                  displayValue: '12 errors',
+                },
+                {
+                  slug: 'no-eval',
+                  title: 'Disallow the use of eval()',
+                  score: 1,
+                  value: 0,
+                  displayValue: 'passed',
+                },
+                {
+                  slug: 'no-var',
+                  title: 'Require `let` or `const` instead of `var`',
+                  score: 0,
+                  value: 5,
+                  displayValue: '5 warnings',
+                },
+              ],
+            },
+          ],
+        },
+        after: {
+          plugins: [
+            {
+              slug: 'eslint',
+              title: 'ESLint',
+              audits: [
+                {
+                  slug: 'eqeqeq',
+                  title: 'Require the use of `===` and `!==`',
+                  // score and value changed
+                  score: 0,
+                  value: 1,
+                  displayValue: '1 error',
+                },
+                // no-case-declarations audit removed
+                {
+                  slug: 'no-eval',
+                  title: 'Disallow the use of eval()',
+                  // unchanged
+                  score: 1,
+                  value: 0,
+                  displayValue: 'passed',
+                },
+                {
+                  slug: 'no-var',
+                  title: 'Require `let` or `const` instead of `var`',
+                  // only value changed
+                  score: 0,
+                  value: 3,
+                  displayValue: '3 warnings',
+                },
+              ],
+            },
+            // plugin added
+            {
+              slug: 'coverage',
+              title: 'Code coverage',
+              audits: [
+                {
+                  slug: 'function-coverage',
+                  title: 'Function coverage',
+                  score: 0.8,
+                  value: 80,
+                  displayValue: '80 %',
+                },
+              ],
+            },
+          ],
+        },
+      } as ReportsToCompare),
```

Comment:
`satisfies` over `as`?

Reply:
It wouldn't compile with `satisfies`, I'm actually intentionally leaving out most `Report` properties because they're not relevant to the test scenario. Using `as` allows me to omit required props, while still giving me autocomplete and some basic type checking - TypeScript treats it as kind of a recursive partial.